.github/workflows/ci.yml

```diff
@@ -82,63 +82,51 @@ jobs:
       - run: ./scalafmt --test
       - run: yarn install
       - run: yarn format-check
-  native-image:
+  scala-native:
     permissions:
       contents: write  # for actions/upload-release-asset to upload release asset
-    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [macOS-latest, ubuntu-latest]
-        libc: [default, musl]
-        exclude:
-          - os: macOS-latest
-            libc: musl
-        include:
-          - os: macOS-latest
-            artifact: scalafmt-macos
-            env:
-              NATIVE_IMAGE_STATIC: none
-          - os: ubuntu-latest
-            libc: default
-            artifact: scalafmt-linux-glibc
-            env:
-              NATIVE_IMAGE_STATIC: nolibc
-          - os: ubuntu-latest
-            libc: musl
-            artifact: scalafmt-linux-musl
-            env:
-              NATIVE_IMAGE_STATIC: musl
-    env: ${{ matrix.env }}
+        deploy: [ 
+          { os : macOS-13, name: scalafmt-x86_64-apple-darwin}, 
+          { os : macOS-14, name: scalafmt-aarch64-apple-darwin}, 
+          { os : ubuntu-latest, name: scalafmt-x86_64-pc-linux}, 
+          { os : ubuntu-24.04-arm, name: scalafmt-aarch64-pc-linux}, 
+          { os : windows-latest, name: scalafmt-x86_64-pc-win32}
+        ]
+    runs-on: ${{ matrix.deploy.os }}
+    env:
+      BINARY_NAME: ${{ (startsWith(matrix.deploy.os, 'windows') && 'scalafmt.exe') || 'scalafmt' }}
     steps:
       - uses: actions/checkout@v4
-      - name: Set up GraalVM
-        uses: graalvm/setup-graalvm@v1
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v4
```

Comment:
a comment for separate consideration: if we are using java 21 for this, should we disable java 8 compatibility in native builds (if they apply, that is)?